### ISSUE TITLE
Markdown card templates using stricter rendering than developer-tools

### ISSUE THREAD

**Issue by erkr** [home-assistant/frontend#20862]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

I have a simple template that filters on the 'device_class' attribute of an entity. That template works fine in e.g dev tools or as an template sensor, but fails in the markdown card for entities that  don't have a 'device_class'.

### Describe the behavior you expected

Template works the same as in Dev Tools or in the template integration. I also verified a number of other cards that support templates, and there it works fine. So it seems a markdown card specific issue.

### Steps to reproduce the issue

1. Created this test template  entity without a device_class:
``` 
- binary_sensor:
  - name: test_source
    state: 'on'
``` 
2. Created this markdown card:
``` 
type: markdown
content: >-
      {% set cnt = expand('binary_sensor.test_source') | rejectattr('attributes.device_class','eq','connectivity') | list | count   -%}
      {{ cnt > 0 }}
``` 
3. Results in this error:
![image](https://github.com/home-assistant/frontend/assets/4363599/3e1db2e2-33f8-41d7-9f12-c5600389aac0)

### What version of Home Assistant Core has the issue?

Core 2024.5.4

### In which browser are you experiencing the issue with?

Chrome, Safari

### Which operating system are you using to run this browser?

Win11 aan iOS 17

**Comment by erkr** [comment]
> Your template seems fine to me in a markdown card:
> 

Thanks for verifying, but it only fails for entities where the device class is missing.

**Comment by karwosts** [comment]
Yes I see what you mean. It seems markdown card subscribes its template with an additional argument `strict: true`, and developer tools does not. 

I am not sure what the reason for this is.

**Comment by karwosts** [comment]
So it seems that `strict` mode for devtools was replaced with an option `report_errors`, but `report_errors` seems to catch less things than strict mode. 

Perhaps that should be reported as a core bug, that report_errors should be able to report everything that strict mode would catch. I assume that was its intention but I don't know.

**Comment by erkr** [comment]
@karwosts 
Nice it seems your identifiers the source of this issue.
But one remark. Rejectattr and selectattr should not raise errors when those attributes don't exist (to my knowledge)?! They will simply not be selected or rejected (like dev tools and template sensors)

**Comment by ildar170975** [comment]
> Rejectattr and selectattr should not raise errors when those attributes don't exist (to my knowledge)?! They will simply not be selected or rejected (like dev tools and template sensors).

My first experiments with selectattr / rejectattr were on 2021 - and on that time I learned that I must check for an attribute existence before using selectattr/rejectattr.

**Comment by erkr** [comment]
> 
> My first experiments with selectattr / rejectattr were on 2021 - and on that time I learned that I must check for an attribute existence before using selectattr/rejectattr. 

How to do that in filter expressions?! When I lookup a value with state_attr(), I have to check, but in a filter I want to include/exclude entities that match. 
I checked several places where I can execute templates (dev tools, template sensors, state switch card etc) and the same filtering works for all, except the markdown card.

**Comment by erkr** [comment]
Other simple example:
``` 
{% set entities = states | rejectattr('attributes.device_class','ne','connectivity')| map(attribute='entity_id') | list    -%}
{{ entities }}
``` 
auto entities card works fine:
![image](https://github.com/home-assistant/frontend/assets/4363599/defcd677-85f8-469c-95ac-d50a652ba0a5)
Markdown card with the same template fails:
![image](https://github.com/home-assistant/frontend/assets/4363599/8dffeda2-c9f7-48ab-9bf7-f21c87ba2d77)

**Comment by erkr** [comment]
Bump
